units/repotracker.go

```diff
@@ -90,6 +90,16 @@ func (j *repotrackerJob) Run() {
 		return
 	}
 
+	if !ref.TracksPushEvents {
```

Comment:
isn't it a programmer error if we run this job for this project?

Reply:
As written, all push events create amboy jobs, and the job is responsible for determining which pushevents should trigger repotracker by looking up the ref. Would you prefer I move the ref validation to the PushEvent handler?

Reply:
I think in that case, it's not really an error? just log a debug and don't even add errors?